### PR TITLE
Barrels & Kegs

### PR DESCRIPTION
**Pull request by rosieposieeee**
## About the PR

Content Client_CZ0fPYsX6l

Content Client_FgX2X3xSbo

Content Client_tvtAOqDXDE

Content Client_RHDsxkBNJQ

Content Client_Col7NY7ccU

Added wooden barrels and kegs. Wine, Beer, and Root Beer kegs are purchasable for 500 spesos individually at cargo.

## Why / Balance

For decoration, and for the bartender to have a keg to drag around that ISN'T identical to a nuke.

## Technical details

Both are fully constructible and deconstructible (though with no SpillAllSolutions completion in constructions, I just used the SolutionEmpty condition to prevent deconstruction of a keg with anything in it). Base wooden barrel sprite adapted from https://github.com/tgstation/tgstation/pull/38977 but redone to fit the style better, plus I sprited the keg wholecloth.

Regular wooden barrel acts like a crate, wooden keg acts like a tank. Keg is set to 500 volume; I felt like it needed to be larger than jugs, but not as large as tanks (which are 1500). Wooden kegs and wooden barrels can be constructed with 5 wood (a la crates), but that can be increased if necessary. They are also vaultable :)

Refilling kegs is a bit annoying, but doable, with the best approach being dragging jugs onto them similar to the chemmaster. You can also inject into them with syringes. Clicking on them with a bottle or anything will just fill that container, similar to a water cooler. You can also drink directly from them.

NOTE: I hereby release all changes made in this PR under MIT license.

## Requirements

- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

**Changelog**
:cl:
- add: Added wooden barrels and wooden kegs.

### CODE REVIEW

**Comment by therealDLondon** [comment]
Very important question, can I fill it with mustard

**Comment by dvir001** [comment]
Funny I had your PR in WIP limbo on Frontier
https://github.com/new-frontiers-14/frontier-station-14/pull/1776

**Comment by Avalon-Proto** [comment]
I have a request- can you make it so we can pick up kegs. But it takes 2 hands like the broken Firelocks, airlocks, and other big salvage items. I wanna be able to just *chuck* a barrel at someone

**Comment by rosieposieeee** [comment]
> I have a request- can you make it so we can pick up kegs. But it takes 2 hands like the broken Firelocks, airlocks, and other big salvage items. I wanna be able to just _chuck_ a barrel at someone

unfortunately that would conflict with vaulting

> Very important question, can I fill it with mustard

if mustard is a reagent, yes

**Comment by Avalon-Proto** [comment]
Already tried it I suppose?

**Comment by rosieposieeee** [comment]
> Already tried it I suppose?

theres really not enough default interaction keys for an object you can vault, pick up, drink from, open/close,

**Comment by Avalon-Proto** [comment]
> theres really not enough default interaction keys for an object you can vault, pick up, drink from, open/close,

To be fair you can just use the context menu too in order to pick em up. Not everything needs to be super quick. Hell lately you have needed to use the context menu to pick up stuff like mice cause you can't just click em anymore

File: Resources/Prototypes/DeltaV/Catalog/Cargo/cargo_food.yml [hunk 1]
```diff
@@ -57,3 +57,33 @@
...
+
+- type: cargoProduct
+  id: FoodWineBarrel
+  icon:
+    sprite: DeltaV/Objects/Storage/keg.rsi
+    state: base
+  product: WoodenKegWine
+  cost: 500
+  category: Food
+  group: market
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Catalog/Cargo/cargo_food.yml]: missing newline at the end

File: Resources/Prototypes/DeltaV/Entities/Structures/Storage/Crates/barrel.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  id: WoodenKegRootBeer
+  components:
+    - type: Label
+      currentLabel: reagent-name-root-beer
+    - type: SolutionContainerManager
+      solutions:
+        tank:
+          reagents:
+            - ReagentId: RootBeer
+              Quantity: 500
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Structures/Storage/Crates/barrel.yml]: for these unindent the components and reagent so - is inline with the field

File: Resources/Prototypes/DeltaV/Recipes/Crafting/Graphs/barrel.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Recipes/Crafting/Graphs/barrel.yml:1]: above

File: Resources/Textures/DeltaV/Objects/Storage/barrel.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from tgstation PR https://github.com/tgstation/tgstation/pull/38977, modified by rosieposieeee",
```
Comment (deltanedas) [inline Resources/Textures/DeltaV/Objects/Storage/barrel.rsi/meta.json]: this pr was merged, why not get a permalink of the dmi on the repo

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Textures/DeltaV/Objects/Storage/barrel.rsi/meta.json (hunk 3 above)
Comment (rosieposieeee) [reply Resources/Textures/DeltaV/Objects/Storage/barrel.rsi/meta.json]: 👍

File: Resources/Prototypes/DeltaV/Entities/Structures/Storage/Crates/barrel.yml (hunk 2 above)
Comment (rosieposieeee) [reply Resources/Prototypes/DeltaV/Entities/Structures/Storage/Crates/barrel.yml]: i think i did this right now... 👍

File: Resources/Prototypes/DeltaV/Recipes/Crafting/Graphs/barrel.yml
Comment (rosieposieeee) [reply Resources/Prototypes/DeltaV/Recipes/Crafting/Graphs/barrel.yml:1]: 👍

File: Resources/Prototypes/DeltaV/Catalog/Cargo/cargo_food.yml (hunk 1 above)
Comment (rosieposieeee) [reply Resources/Prototypes/DeltaV/Catalog/Cargo/cargo_food.yml]: 👍

**Review by Lyndomen — APPROVED** [review]
Every day we borrow inspiration from dwarf fortress is a good day

**Review by deltanedas — APPROVED** [review]
ill clean up later

**Comment by jimmy12or** [comment]
I love this. I will likely use this as a botanist to offer special juices!

**Pull request merged**